certification/internal/cli/openshift.go

```diff
@@ -37,21 +35,23 @@ type OperatorGroupData struct {
 }
 
 type OpenshiftEngine interface {
-	CreateNamespace(name string, opts OpenshiftOptions, config *rest.Config) (*OpenshiftReport, error)
-	DeleteNamespace(name string, opts OpenshiftOptions, config *rest.Config) error
-	GetNamespace(name string, config *rest.Config) (*corev1.Namespace, error)
+	Setup() error
 
-	CreateOperatorGroup(data OperatorGroupData, opts OpenshiftOptions, config *rest.Config) (*OpenshiftReport, error)
-	DeleteOperatorGroup(name string, opts OpenshiftOptions, config *rest.Config) (*OpenshiftReport, error)
-	GetOperatorGroup(name string, opts OpenshiftOptions, config *rest.Config) (*operatorv1.OperatorGroup, error)
+	CreateNamespace(name string, opts OpenshiftOptions) (*OpenshiftReport, error)
```

Comment:
It appears that the OpenShiftReport is being ignored for the most part. I think we need to think about this interface a bit more. OpenshiftReport doesn't make sense (with Stdout, and Stderr), since that's not what the implementation is meant to do anyway. We aren't using an external cli here. What should be returned from a Create? The entiry that was created? Just the error?

Reply:
Looking at the fake engine that you created, it does seem like Create should return the object, and an error. The deletes are inconsistent... Some just return error, others return the OpenshiftReport (which again, doesn't really make sense).

Reply:
Thank you Brad for the feedbacks. I am with you on removing the OpenShiftReport all together and redesign the interface. My initial intention was to be consistent with the other engines, but we probably don't need this for Openshift. I am open to discuss this further.

Reply:
I'm fine with going with it as is, and iterating on it. I think the longer-term design shouldn't be rushed.

Reply:
I removed the OpenshiftReport type and made the methods consistent. We will add it back if needed.